### PR TITLE
jekyll build automation using TravisCI

### PR DESCRIPTION
**Pull request by kinueng**
Fixes kappnav/issues#32

Travis will run the `jekyll build` process, and push the `_site` folder content to the `master` branch.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  github_token: $GITHUB_TOKEN
+  local_dir: _site
+  target_branch: master
+  on:
+    branch: source
+
+# branch whitelist, only for GitHub Pages
+branches:
+  only:
+  - source
```
Comment (erica-banda-03) [inline .travis.yml]: Remove this so we can build all branches

File: README.md [hunk 2]
```diff
@@ -0,0 +1 @@
+[![Build Status](https://travis-ci.com/kappnav/kappnav.github.io.svg?branch=source)](https://travis-ci.com/kappnav/kappnav.github.io)
```
Comment (erica-banda-03) [inline README.md]: Should add all relevant badges like the Apache license

File: _config.yml [hunk 3]
```diff
@@ -34,14 +34,12 @@ assets:
...
-#   - vendor/cache/
-#   - vendor/gems/
-#   - vendor/ruby/
+# Travis bundles all gems in the vendor directory on its build servers, 
+# which Jekyll will mistakenly read and explode on.
+exclude:
+  - vendor 
+  - Gemfile
+  - Gemfile.lock
+  - LICENSE
```
Comment (erica-banda-03) [inline _config.yml]: We don't have this file in source, please remove.

File: _config.yml [hunk 4]
```diff
@@ -34,14 +34,12 @@ assets:
...
+  - README.md
+  - Rakefile
```
Comment (erica-banda-03) [inline _config.yml]: We don't have this file in source, please remove.

**Comment by kinueng** [comment]
@erica-banda-03, Please review my recent changes to this PR.

Turns out we need to migrate a few files from `master` to `source`.  (e.g. LICENSE).  TravisCI does a `git push --force` from it pushes to `master`, causing the loss of those files.

I also made some changes so that we can download the website source and load it with only using the files from the zip containing the website source.

I did all my testing on a fork https://github.com/kinueng/kappnav.github.io/tree/master

**Comment by erica-banda-03** [comment]
Let's try the flag in Travis to retain git history on the master branch

**Pull request merged**